specification/deviceregistry/DeviceRegistry.Management/schemaRegistries.tsp

```diff
@@ -0,0 +1,205 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "./common/provisioningState.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.ResourceManager;
+using OpenAPI;
+
+namespace Microsoft.DeviceRegistry;
+
+@added(Versions.v2024_09_01_preview)
+@doc("Schema registry definition.")
+model SchemaRegistry is TrackedResource<SchemaRegistryProperties> {
+  @doc("Schema registry name parameter.")
+  @key("schemaRegistryName")
+  @segment("schemaRegistries")
+  @pattern("^[a-z0-9][a-z0-9-]*[a-z0-9]$")
+  @path
+  @minLength(3)
+  @maxLength(63)
+  name: string;
+
+  ...ManagedSystemAssignedIdentityProperty;
+}
+
+@added(Versions.v2024_09_01_preview)
+@doc("Defines the schema registry properties.")
+model SchemaRegistryProperties {
+  @doc("Globally unique, immutable, non-reusable id.")
+  @visibility("read")
+  uuid?: string;
+
+  @doc("Schema registry namespace. Uniquely identifies a schema registry within a tenant.")
+  @visibility("read", "create")
+  @pattern("^[a-z0-9][a-z0-9-]*[a-z0-9]$")
+  @maxLength(32)
+  `namespace`: string;
+
+  @doc("Human-readable display name.")
+  displayName?: string;
+
+  @doc("Human-readable description of the schema registry.")
+  description?: string;
+
+  @doc("The Storage Account's Container URL where schemas will be stored.")
+  @visibility("read", "create")
+  storageAccountContainerUrl: string;
+
+  @doc("Provisioning state of the resource.")
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+}
+
+@added(Versions.v2024_09_01_preview)
+@parentResource(SchemaRegistry)
+@doc("Schema definition.")
+model Schema is ProxyResource<SchemaProperties> {
+  @doc("Schema name parameter.")
+  @key("schemaName")
+  @segment("schemas")
+  @pattern("^[a-z0-9][a-z0-9-]*[a-z0-9]$")
+  @path
+  @minLength(3)
+  @maxLength(64)
+  name: string;
+}
+
+@added(Versions.v2024_09_01_preview)
+@doc("Defines the schema properties.")
+model SchemaProperties {
+  @doc("Globally unique, immutable, non-reusable id.")
+  @visibility("read")
+  uuid?: string;
+
+  @doc("Human-readable display name.")
+  displayName?: string;
+
+  @doc("Human-readable description of the schema.")
+  description?: string;
+
+  @doc("Format of the schema.")
+  @visibility("read", "create")
+  format: Format;
+
+  @doc("Type of the schema.")
+  @visibility("read", "create")
+  schemaType: SchemaType;
+
+  @doc("Provisioning state of the resource.")
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" ""
+  @doc("Schema tags.")
+  // Allowed scenario as per user-defined tags: https://github.com/Azure/azure-openapi-validator/blob/main/docs/avoid-additional-properties.md
+  tags?: Record<string>;
```

Comment:
Proxy resources don't support tags

Reply:
@davidemontanari I believe this is related to the `TagsAreNotAllowedForProxyResources` violations. You need to make them tracked resources if you need tags support.

Reply:
@gary-x-li, `tags` here is defined inside the resource `properties` (accessed through `properties.tags`) and not as top level property. Tracked resources define `tags` as resource top level property (outside `properties`). Is that considered the same?

Reply:
Thanks for the explanation. It's different from the top level `tags` so it's fine here.